src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslDirectoryBasedStoreProvider.cs

```diff
@@ -187,9 +202,15 @@ private void AddCertToStore(ICertificatePal certPal)
                     destinationFilename = Path.Combine(_storePath, thumbprint + PfxExtension);
                 }
 
-                using (FileStream stream = new FileStream(destinationFilename, mode))
+                using (FileStream stream = new FileStream(destinationFilename, options))
                 {
-                    EnsureFilePermissions(stream, userId);
+                    // Verify the file can only be read/written to by the owner.
+                    UnixFileMode actualMode = File.GetUnixFileMode(stream.SafeFileHandle);
+                    if (actualMode != UserReadWrite)
+                    {
+                        throw new CryptographicException(SR.Format(SR.Cryptography_InvalidFilePermissions, stream.Name));
+                    }
```

Comment:
note: I've omitted the owner check that was in `EnsureFilePermissions` since it is implicit from only the owner having permissions and us being able to open the file.